### PR TITLE
feat: add module logic

### PR DESCRIPTION
**Pull request by kolbeinn**
## Summary

This PR contains the core logic, documentation, and examples for this Terraform module. This module is meant to contain the reusable logic regarding user/group creation that can be reused across both the configuration integration module and the audit log integration module.

## How did you test this change?

I ran all the examples in the `examples` folder:

```bash
cd examples/<example>
tf init
TF_VAR_tenancy_id=<tenancy_id> tf plan
TF_VAR_tenancy_id=<tenancy_id> tf apply
TF_VAR_tenancy_id=<tenancy_id> tf destroy
```

## Issue

[SPM-767](https://lacework.atlassian.net/browse/SPM-767)

[SPM-767]: https://lacework.atlassian.net/browse/SPM-767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+
+variable "freeform_tags" {
+  type        = map(any)
+  default     = {}
+  description = "freeform tags for the resources created for Lacework integration"
+}
+
+variable "email" {
+  type        = string
+  default     = "lacework@lacework.net"
```
Comment (iok0) [inline variables.tf:24]: Maybe we should have no default for email since this won't be monitored?

**Review by iok0 — APPROVED** [review]
LGTM

File: variables.tf (hunk 1 above)
Comment (kolbeinn) [reply variables.tf:24]: Good idea. For some reason, I thought the email was a required parameter, but I see now that it's an optional parameter. Will remove it.

Comment (kolbeinn) [reply variables.tf:24]: Turns out it is a required parameter. The Terraform docs say it's optional, but the OCI API returns an error (""The primary email must be specified") if you don't specify an email. Leaving it as an empty string the same error.

**Pull request merged**